packages/web3-react/src/hooks/useInjectedProviders.tsx

```diff
@@ -16,19 +16,12 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { AlephiumWindowObject } from '@alephium/get-extension-wallet'
-import { useEffect, useState } from 'react'
+import { useSyncExternalStore } from 'use-sync-external-store/shim'
 import { injectedProviderStore } from '../utils/providers'
 
-export function useInjectedProviders() {
-  const [providers, setProviders] = useState<AlephiumWindowObject[]>(injectedProviderStore.getProviders())
-
-  useEffect(() => {
-    const cancel = injectedProviderStore.subscribe(setProviders)
-    return () => {
-      cancel()
-    }
-  }, [setProviders])
-
-  return providers
-}
+export const useInjectedProviders = () =>
```

Comment:
Even though we check if the provider already exists when adding, sometimes duplicate providers appear when refreshing the page due to caching. This issue does not occur in `react-dapp-template` or `nextjs-app-dapp-template` but sometimes occurs in bridge ui.

In the latest commit, `useSyncExternalStore` will be used instead of `useEffect/useState`. The `useSyncExternalStore` hook was introduced in React 18, but to maintain compatibility with React 17, we need to import it from `use-sync-external-store/shim`, which is an official library provided by React.